### PR TITLE
2 feature 1 user login system 1.1

### LINKED ISSUE #2: [Feature 1] <User Login System>
A UserLogin System that allows new users to register and current users to sign in

### PR DESCRIPTION
**Pull request by ghost**
This PR implemented the controller and presenter classes for the 3 uses cases Some part of the view model in the presenters hasn't been updated for other features of the projects such as moving on to the dashboard after logging in and some major refactoring of the code is still needed
with Javadocs, coding styles.

### CODE REVIEW

**Review by TimmyTurner691 — DISMISSED** [review]
What would be great is if there was some java documentation so that the code could be easier to understand but we can worry about that another day I think. Other than that the code is pretty clean, it separates all the use cases into its own class, that way the code doesn't get cluttered from implementing to many classes in one file.

File: src/main/java/controller/Controller.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import java.sql.Statement;
+import ui.ViewLoginPage;
+import usecases.UseCaseFalseLogin;
+import usecases.UseCaseLogin;
+import usecases.UseCaseRegister;
+
+public class Controller {
+    /**
+     * The Controller for the loginButton takes in inputs from the text fields
+     * of the given UI page, and determine which use case to select.
```
Comment (jhalaksaraogi) [inline src/main/java/controller/Controller.java:18]: Might be a good idea to add @param in the javadoc for methods with parameters

File: src/main/java/controller/Controller.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        else{
+            UseCaseRegister.newUser(username, password, email);
+            UseCaseRegister.Registered(page);
+        }
+    }
+
+    /**
+     * This method determine whether the given input from the UI to the username
+     * text field exists in the database
+     *
```
Comment (jhalaksaraogi) [inline src/main/java/controller/Controller.java:58]: Again adding an @return tag to the javadoc for methods that return something  might be a good idea

File: src/main/java/entity/EntitiyAppUser.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package entity;
+public class EntitiyAppUser implements UserDetails {
```
Comment (jhalaksaraogi) [inline src/main/java/entity/EntitiyAppUser.java:2]: Add javadoc to this class

File: src/main/java/usecases/UseCaseRegister.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    /**
+     * This method calls the usernameExists method from the presenter class if the username already exists when the user try signing up
+     */
+    public static void usernameExists(ViewLoginPage x){
+        Presenter.usernameExists(x);
+    }
+
+}
```
Comment (jhalaksaraogi) [inline src/main/java/usecases/UseCaseRegister.java:49]: All javadocs are clear and detailed. Just adding @return and @param tags when relevant might make the documentation more organized and defined.

**Review by bdls-jamal — COMMENTED** [review]
Test Coverage: 
- need to add test cases
- test for edge cases

**Review by bdls-jamal — APPROVED** [review]
